src/test/java/org/fogbowcloud/manager/core/plugins/cloud/openstack/OpenStackV2NetworkPluginTest.java

```diff
@@ -76,46 +88,261 @@ public void validate() {
 		Mockito.validateMockitoUsage();
 	}
 
+	//requestInstance tests
+
+	//test case: The http client must make only 5 requests
+	@Test
+	public void testNumberOfRequestsInSucceededRequestInstance() throws IOException, FogbowManagerException, UnexpectedException {
+		//set up
+		// post network
+		Mockito.doReturn(NETWORK_ID).when(this.httpRequestClientUtil)
+				.doPostRequest(Mockito.endsWith(OpenStackV2NetworkPlugin.SUFFIX_ENDPOINT_NETWORK),
+						Mockito.eq(this.defaultToken), Mockito.any(JSONObject.class));
+		//post subnet
+		Mockito.doReturn("").when(this.httpRequestClientUtil)
+				.doPostRequest(Mockito.endsWith(OpenStackV2NetworkPlugin.SUFFIX_ENDPOINT_SUBNET),
+						Mockito.eq(this.defaultToken), Mockito.any(JSONObject.class));
+		//post security group
+		Mockito.doReturn("").when(this.httpRequestClientUtil)
+				.doPostRequest(Mockito.endsWith(OpenStackV2NetworkPlugin.SUFFIX_ENDPOINT_SECURITY_GROUP),
+						Mockito.eq(this.defaultToken), Mockito.any(JSONObject.class));
+		Mockito.doReturn(SECURITY_GROUP_ID).when(this.openStackV2NetworkPlugin).getSecurityGroupIdFromPostResponse(Mockito.anyString());
+		//post ssh rule
+		Mockito.doReturn("").when(this.httpRequestClientUtil)
+				.doPostRequest(Mockito.endsWith(OpenStackV2NetworkPlugin.SUFFIX_ENDPOINT_SECURITY_GROUP_RULES),
+						Mockito.eq(this.defaultToken), Mockito.any(JSONObject.class));
+		//post icmp rule
+		Mockito.doReturn("").when(this.httpRequestClientUtil)
```

Comment:
Do we need this duplication? Or is line 111 already specifying this?

Reply:
Each security group rule requires a new request. So, when we create a security group we need to add 2 new rules, allow SSH and ICMP, that's the reason for the 'duplicated' code, we're mocking both requests.

Reply:
Fixed, thanks.